nixos/modules/services/network-filesystems/s3backer.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.s3backer;
+  mountOption = {name, ...}: {options = {
+    accessFile = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = ''
+       Specify a file containing `accessID:accessKey' pairs, one per-line.  Blank lines and lines beginning with a `#' are ignored.  If no
+               --accessKey or --accessKeyEnv is specified, this file will be searched for the entry matching the access ID specified via --accessId; if no
+               --accessId is specified, the first entry in this file will be used.  If not specified, $HOME/.s3backer_passwd will be used.
+      '';
+    };
+    bucketPath = mkOption {
+      type = types.str;
+      description = ''
+        Bucket and optional key within bucket to create backing file for.
+      '';
+      example = "bucket_name/subdir";
+    };
+    encrypt = mkOption {
+      default = false;
+      type = types.nullOr types.bool;
+      description = ''
+        Enable encryption and authentication of block data.
+      '';
+    };
+    passwordFile = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = ''
+        File containing the encryption password for the 'encrypt' option.
+      '';
+    };
+    mountPoint = mkOption {
+      default = "/var/lib/s3backer/${name}";
+      type = types.str;
+      description = "Mount point for s3backer FUSE filesystem";
+    };
+    size = mkOption {
+      type = types.str;
+      description = ''
+        Specify the size (in bytes) of the backed file to be exported by the filesystem.  The size may have an optional suffix 'K' for kilobytes, 'M'for megabytes, 'G' for gigabytes, 'T' for terabytes, 'E' for exabytes, 'Z' for zettabytes, or 'Y' for yottabytes.
+      '';
+    };
+    filename = mkOption {
+      default = "file";
+      type = types.str;
+      description = ''
+        the name of the backed file that appears in the s3backer filesystem.
+      '';
+    };
+    extraOptions = mkOption {
+      default = null;
+      type = types.nullOr types.lines;
+      description = ''
+        Extra command-line arguments to be passed to s3backer.
+
+        Leading and trailing whitespace is trimmed from each line, and blank lines and lines starting with # are ignored.  Each line contains exactly
+             one command line argument (including internal whitespace, if any).
+      '';
+    };
+  };};
+in {
+  options.services.s3backer = {
+    package = mkPackageOption pkgs "s3backer" {};
+    mounts = mkOption {
+      default = {};
+      type = types.attrsOf (types.submodule mountOption);
+      description = ''
+        Configuration for s3backer mount points.
+      '';
+    };
+  };
+  config.systemd.services = mapAttrs' (name: mount:
+    let options = [
```

Comment:
I feel like there's a more idiomatic way to do this but I don't know what it is. I'd appreciate suggestions on how to improve this.